### PR TITLE
webauthn cardinality fix

### PR DESCRIPTION
**Pull request by magizh-okta**
Commit 1 -  Fixes to accomodate updates to api due to factor profiles.
Already reviewed 
https://github.com/okta/okta-signin-widget/pull/729

commit 2 -  Fix cardinality text when max is 1. With factor profiles cardinality is always present. Max 1 means the factor actually doenst support cardinality. 
**Before** 
Screen Shot 2019-08-06 at 5 12 58 PM
**After**
Screen Shot 2019-08-06 at 5 14 50 PM

More testing videos 
**Multiple webauthn optional** 
mul-web
**multiple webauthn from enduser**
mul-web2-enduser-settingw
**multiple webauthn required** 
mul-web3-req
**claims provider  required** 
mul-web4-idp-required-real
**claims optional** 
mul-web4-idp
**totp** 
mul-web4-totp-required

**Max enrolled**
Screen Shot 2019-08-08 at 9 46 43 AM

### CODE REVIEW

**Comment by haishengwu-okta** [comment]
Please bump the version in a separated commit.

**Comment by magizh-okta** [comment]
@haishengwu-okta as discussed I am not updating the commit and @manueltanzi-okta  (widget co-ordinator) would be doing it.

File: test/unit/spec/EnrollChoices_spec.js [hunk 1]
```diff
@@ -924,6 +924,20 @@ function (Okta, OktaAuth, Util, EnrollChoicesForm, Beacon, Expect, FactorUtil, R
             expect(test.form.factorCardinalityText(factorName)).toBe('');
           });
         });
+        itp('does not display cardinality text if it is max is 1', function () {
```
Comment (gowthamidommety-okta) [inline test/unit/spec/EnrollChoices_spec.js]: nit: `does not display cardinality text when  factor is REQUIRED and maximum=1`

File: test/unit/spec/EnrollChoices_spec.js [hunk 2]
```diff
@@ -880,232 +916,348 @@ function (Okta, OktaAuth, Util, EnrollChoicesForm, Beacon, Expect, FactorUtil, R
...
+      function testMultipleEnrollmentsForFactor (factorName, webauthnEnabled, useProfiles, enrolledFactor) {
         itp('does not display cardinality text if it is optional', function () {
-          return setup(resAllFactors).then(function (test) {
+          var res = useProfiles ? resAllFactorsProfile : resAllFactors;
+          return setup(res, false, webauthnEnabled).then(function (test) {
             expect(test.form.factorButtonText(factorName)).toBe('Setup');
             expect(test.form.factorCardinalityText(factorName)).toBe('');
           });
         });
+        itp('does not display cardinality text if it is max is 1', function () {
```
Comment (mauriciocastillosilva-okta) [inline test/unit/spec/EnrollChoices_spec.js]: minor corrections: `does not display cardinality text if maximum=1`

**Pull request merged**